waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -26,15 +24,16 @@ import qualified Wasp.Util.Terminal as Term
 newtype ProjectName = ProjectName {_projectName :: String}
 
 createNewProject :: String -> Command ()
-createNewProject (all isLetter -> False) =
-  throwError $ CommandError "Project creation failed" "Please use only letters for a new project's name."
-createNewProject ((`elem` reservedNames) -> True) =
-  throwError $
-    CommandError
-      "Project creation failed"
-      ( "Please pick a project name not one of these reserved words:\n\t" ++ intercalate "\n\t" reservedNames
-      )
-createNewProject name = createNewProject' (ProjectName name)
+createNewProject name
+  | not (isValidWaspIdentifier name) =
+      throwError $ CommandError "Project creation failed" "The project's name must be a valid Wasp identifier."
+  | name `elem` reservedNames =
```

Comment:
I think we don't have reserved names anymore actually! That is part of the old Parser I think. Can you check? Where are these reserved names coming from?

Reply:
They're coming from the old Lexer (`Wasp.Lexer`) and this is the only place they're used.

Oh, I see, the new parser doesn't parse reserved words as identifier tokens, it parses them as their own thing. In other words, the first guard will take care of it?

If that's the case, then great! I can remove the legacy Lexer

Reply:
Yeah, remove it all!
So new parser doesn't care that much about reserved words really, I think you can even nam the app as `app app {...}`. So I think we can just drop this check, and if needed in the future we will return it in some way.

Reply:
What about the stuff that's between the lines 52 and 62 in `Parser.y`?

Btw, if we really don't have any reserved words, should we consider adding some? Anticipating future reserved words seems pretty common when designing languages. For example, [Java did it with `goto`](https://stackoverflow.com/a/2545141) because "goto is bad" was a more controversial opinion back then. JavaScript had a huge list of future reserved words and they ended up needing a lot of them ([these](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#future_reserved_keywords) are the only unused reserved words remaining).

Definitely not something we should be worrying about in alpha (or even beta), but interesting nonetheless.